src/modeler/modeler/Modeler.cpp

```diff
@@ -144,38 +142,52 @@ void Modeler::run() const
       0};
 
     // Sub problem
-    auto masterAndSubPbvars = std::make_unique<MasterAndSubPbVariables>();
-    std::string pbId = "1-1";
-    masterAndSubPbvars->setProblemIdentifier(pbId);
+    MasterAndSubPbVariables masterAndSubPbvars;
 
-    OrtoolsLinearProblem ortools_linear_problem(isMip, parameters.solver);
+    OrtoolsLinearProblem subproblem(isMip, parameters.solver);
 
     // gp : class SystemLinearProblemBuilder should be renamed into ComponentFillersBuilder
     // gp : and build() should return the vector of component fillers
-    SystemLinearProblemBuilder system_linear_problem(data.system.get(),
-                                                     ortools_linear_problem,
-                                                     *data.dataSeries,
-                                                     data.scenarioGroupRepository,
-                                                     masterAndSubPbvars.get());
+    // Subproblem
+    SystemLinearProblemBuilder subproblem_builder(data.system.get(),
```

Comment:
I would expect that we first build + dump the **master problem**, and then build + dump+ solve the **sub-problem**.